FIAT/polynomial_set.py

```diff
@@ -176,15 +176,51 @@ def polynomial_set_union_normalized(A, B):
     not contain any of the same members of the set, as we construct a
     span via SVD.
     """
-    new_coeffs = numpy.concatenate((A.coeffs, B.coeffs), axis=0)
+    assert A.get_reference_element() == B.get_reference_element()
+    new_coeffs = construct_new_coeffs(A.get_reference_element(), A, B)
+
+    deg = max(A.get_degree(), B.get_degree())
+    em_deg = max(A.get_embedded_degree(), B.get_embedded_degree())
     coeffs = spanning_basis(new_coeffs)
     return PolynomialSet(A.get_reference_element(),
-                         A.get_degree(),
-                         A.get_embedded_degree(),
+                         deg,
+                         em_deg,
                          A.get_expansion_set(),
                          coeffs)
 
 
+def construct_new_coeffs(ref_el, A, B):
+    # Constructs new coefficients for the set union of A and B
+    # If A and B do not have the same degree the smaller one
+    # is extended to match the larger
+
+    sd = ref_el.get_spatial_dimension()
+    if A.get_embedded_degree() != B.get_embedded_degree():
```

Comment:
We need to check also for the continuity of the expansion sets. We throw a ValueError if they are different. This logic would only work for discontinuous expansion sets, and break if the expansion set continuity is "C1".
```suggestion
    if A.get_embedded_degree() != B.get_embedded_degree() and continuity is None:
```